static/js/containers/LearnerPage.js

```diff
@@ -105,18 +139,34 @@ class LearnerPage extends React.Component<*, LearnerPageProps, *> {
       children,
       profileProps,
       email,
-      openEmailComposer
+      openEmailComposer,
+      coupons,
+      prices,
     } = this.props;
 
     let profile = {};
     let toRender = null;
     let loaded = false;
+    let priceLoaded = false;
+    let couponsLoaded = false;
+    let profileLoaded = false;
+    let isPrivileged = this.isPrivileged(username);
+
     if (profiles[username] !== undefined) {
       profile = profiles[username];
-      loaded = profiles[username].getStatus !== FETCH_PROCESSING;
+      profileLoaded = profiles[username].getStatus !== FETCH_PROCESSING;
+      priceLoaded = R.or(
+        !isPrivileged, // that mean he can not download price of other leaner, so we skip checking loading
+        !R.isEmpty(prices) && prices[username].getStatus !== FETCH_PROCESSING
+      );
```

Comment:
instead of this, could we do something like what we do for the dashboard, below? we have a function (`getFocusedDashboard`) which returns the dashboard if you're a privileged user, and an empty object if not. In `Learner.js` we check if those are empty or not before we show the `StaffLearnerInfoCard`.